src/outlet/index.ts

```diff
@@ -214,6 +327,48 @@ export class Outlet {
 		window.dispatchEvent(event)
 	}
 
+	async sendMultiTokenProof(evtid: string, tokens: string, address: string, wallet: string) {
+		if (!tokens) return 'error'
+
+		const redirect = this.getDataFromQuery('redirect') === 'true' ? window.location.href : false
+
+		try {
+			const tokensObj = JSON.parse(tokens)
+
+			let output = { issuers: {}, issuersValidated: [] }
+
+			await Promise.all(
+				Object.keys(tokensObj).map(async (key) => {
+					if (!output.issuers[key]) {
+						output.issuers[key] = []
+						output.issuersValidated.push(key)
+					}
+					for (const token of tokensObj[key].requestTokens) {
+						const issuer = this.getIssuerConfigById(key)
+						const ticketRecord = await this.ticketStorage.getStoredTicketFromDecodedToken(
+							createIssuerHashArray(issuer),
+							token.unsignedToken,
+						)
+						let authHandler = new AuthHandler(issuer, ticketRecord, tokensObj[key].unsignedToken, address, wallet, redirect, this, evtid)
+						let tokenProof = await authHandler.authenticate()
```

Comment:
@nicktaras in Chrome it case if email attestation is not ready mutiple simultaneous authenticate() will open multiple iframes or rewrite each other. anyway problem can happen.

Reply:
From testing this in the console, it appears to do everything nicely in order. This is a very complex process for the browser to make here - My preference after this MVP solution is to move this to attestation.id to resolve all email attestations and allow the TKN to process them - this will be a much more user friendly experience also (keeping the user in one place at a time).

Reply:
Thoughts are we can change this to trigger a singleton class, and have the method receive the given params,
but this would be more likely to have code re-writing each other in a scenario like this - my understanding is that Miccy had taken this approach upfront to allow for many classes to sit in memory, such as Auth handler and some others that use this idea - instantiating class state and having methods with no params to use the class state data etc?

Reply:
its easy to fix by running authenticate() one by one, syncroniously. just add for() and let tokenProof = await authHandler.authenticate() inside loop @nicktaras

Reply:
Done. Auth handler has been renamed to attestationIdClient and will store ID attestations in local storage to support redirect mode.